### PR TITLE
XRE-1791

### PR DESCRIPTION
**Pull request by ramang-unity**
Now SRP can turn on STEREO_MULTIVIEW_ON if STEREO_INSTANCING_ON is on and uses Vulkan.

### **Please read**
**PR workflow guidelines**
* SRP ABV will start automatically on Yamato when you open your PR
* Changes to docs and md files will **not** trigger ABV jobs 
* Consider making use of **draft PRs** if you are not 100% sure that your PR is ready for review
* ABV will restart if you add a new commit to a branch with an open PR (hence why you should consider using draft PRs)
* Adding [skip ci] (case insensitive) to the title of PRs will stop any jobs being trigger automatically - you will need to open Yamato and find your branch to run ABV
* You can also add [skip ci] to commit messages to prevent CI from running on that push
* Add [cancel old ci] to your commit message if you've made changes you want to test and no longer need the previous jobs

### Checklist for PR maker
- [ ] Have you added a backport label (if needed)? For example, the `need-backport-*` label. After you backport the PR, the label changes to `backported-*`.
- [ ] Have you updated the changelog? Each package has a `CHANGELOG.md` file.
- [ ] Have you updated or added the documentation for your PR? When you add a new feature, change a property name, or change the behavior of a feature, it's best practice to include related documentation changes in the same PR.
- [ ] Have you added a graphic test for your PR (if needed)? When you add a new feature, or discover a bug that tests don't cover, please add a graphic test.

---
### Purpose of this PR
Why is this PR needed, what hard problem is it solving/fixing?

---
### Testing status

**Manual Tests**: What did you do?
- [ ] Opened test project + Run graphic tests locally
- [ ] Built a player
- [ ] Checked new UI names with UX convention
- [ ] Tested UI multi-edition + Undo/Redo + Prefab overrides + Alignment in Preset
- [ ] C# and shader warnings (supress shader cache to see them)
- [ ] Checked new resources path for the reloader (in developer mode, you have a button at end of resources that check the paths)
- Other: 

**Automated Tests**: What did you setup? (Add a screenshot or the reference image of the test please)

**Yamato**: (Select your branch):
https://yamato.prd.cds.internal.unity3d.com/jobs/902-Graphics

Any test projects to go with this to help reviewers?

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

File: com.unity.render-pipelines.universal/Editor/ShaderPreprocessor.cs [hunk 1]
```diff
@@ -237,7 +240,23 @@ public void OnProcessShader(Shader shader, ShaderSnippetData snippetData, IList<
...
+                        ShaderCompilerData multiViewData = new ShaderCompilerData();
+                        ShaderKeywordSet keyWordSet;
+                        keyWordSet = compilerDataList[i].shaderKeywordSet;
+                        keyWordSet.Disable(m_SinglePassInstance);
+                        keyWordSet.Enable(m_Multiview);
+                        multiViewData.platformKeywordSet = compilerDataList[i].platformKeywordSet;
+                        multiViewData.shaderKeywordSet = keyWordSet;
+                        multiViewData.graphicsTier = compilerDataList[i].graphicsTier;
+                        multiViewData.shaderCompilerPlatform = compilerDataList[i].shaderCompilerPlatform;
+                        multiViewData.shaderRequirements = compilerDataList[i].shaderRequirements;
```
Comment (ramang-unity) [inline com.unity.render-pipelines.universal/Editor/ShaderPreprocessor.cs:255]: This requires a change on the Unity side. Would this be okay?

**Comment by phi-lira** [comment]
Instead of patching keyword sets on the preprocessor can't we define STEREO_MULTIVIEW_ON if vulkan and stereo instaning is enabled in the shader?

**Comment by ramang-unity** [comment]
> Instead of patching keyword sets on the preprocessor can't we define STEREO_MULTIVIEW_ON if vulkan and stereo instaning is enabled in the shader?

Oh this is for the case if the shader has multi_compile and both SPI and Multiview are variants in the shader. In the Unity code it checks platform and if the platform supports multiview it will remove SPI and just add Multiview. This reduces build time of shader variants. I'm moving this over to SRP to control instead of it being hidden in Unity and difficult to change
This is just draft/hack to prove it can be done. I'm adding API to allow SRP to check if the build target supports SPI or Multiview during build time of shaders.

**Comment by thomas-zeng** [comment]
Hey Raman, could you make this a draft PR if this is not intend to land?

**Comment by ramang-unity** [comment]
> Hey Raman, could you make this a draft PR if this is not intend to land? Thank you!

It already is, no?

**Comment by thomas-zeng** [comment]
It is already a draft PR. Sorry to bother. :)

**Comment by ramang-unity** [comment]
@aleks01010101 I know. Just adding you to see changes in SRP :)

**Comment by phi-lira** [comment]
What's the status on this PR?

**Comment by ramang-unity** [comment]
@phi-lira this PR is on hold because of how we want to implement the changes on trunk and UI.

**Comment by phi-lira** [comment]
Update on this PR?

**Pull request closed without merging**